crates/sui-adapter/src/adapter.rs

```diff
@@ -557,26 +555,29 @@ fn process_successful_execution<
                                 ctx.sender(),
                                 *obj_id,
                             ));
-                            let previous_version =
-                                match state_view.get_latest_parent_entry_ref(*obj_id) {
-                                    Ok(Some((_, previous_version, _))) => previous_version,
-                                    // TODO this error is (hopefully) transient and should not be
-                                    // a normal execution error
-                                    _ => {
-                                        return Err(ExecutionError::new_with_source(
-                                            ExecutionErrorKind::InvariantViolation,
-                                            missing_unwrapped_msg(obj_id),
-                                        ));
-                                    }
-                                };
-                            unwrap_and_deleted.insert(*obj_id);
-                            changes.insert(
-                                *obj_id,
-                                ObjectChange::Delete(
-                                    previous_version,
-                                    DeleteKind::UnwrapThenDelete,
-                                ),
-                            );
+                            match state_view.get_latest_parent_entry_ref(*obj_id) {
+                                Ok(Some((_, previous_version, _))) => {
+                                    changes.insert(
+                                        *obj_id,
+                                        ObjectChange::Delete(
+                                            previous_version,
+                                            DeleteKind::UnwrapThenDelete,
+                                        ),
+                                    );
+                                }
+                                // if the object is not in parent sync, it was wrapped before
+                                // ever being stored into storage. Thus we don't need to mark it
+                                // as being deleted
+                                Ok(None) => (),
+                                // TODO this error is (hopefully) transient and should not be
+                                // a normal execution error
+                                Err(_) => {
+                                    return Err(ExecutionError::new_with_source(
+                                        ExecutionErrorKind::InvariantViolation,
```

Comment:
This should be a system error? Or is it because we don't yet have the framework for handling system error in execution. Would be good to link to an issue here.

Reply:
Do we have an issue for it?

Reply:
(filed one)